tests/sentry/api/endpoints/test_project_details.py

```diff
@@ -251,120 +252,145 @@ class ProjectUpdateTestTokenAuthenticated(APITestCase):
+            authorization = self.create_bearer_token(self.user, [scope])
 
-        data = {"platform": "rust"}
+            response = self.client.put(
+                self.url, format="json", HTTP_AUTHORIZATION=authorization, data=self.data
+            )
+            assert response.status_code == 200, response.content
 
-        url = reverse(
-            "sentry-api-0-project-details",
-            kwargs={
-                "organization_slug": self.project.organization.slug,
-                "project_slug": self.project.slug,
-            },
+    def test_manager_update_allowed_with_correct_token_scope(self):
+        self.create_member(
+            user=self.user,
+            organization=self.project.organization,
+            teams=[self.team],
+            role="manager",
         )
-        response = self.client.put(url, format="json", HTTP_AUTHORIZATION=authorization, data=data)
-        assert response.status_code == 403, response.content
 
-    def test_admin_updates_allowed_with_correct_token_scope(self):
+        for scope in ["project:write", "project:admin"]:
+            authorization = self.create_bearer_token(self.user, [scope])
+
+            response = self.client.put(
+                self.url, format="json", HTTP_AUTHORIZATION=authorization, data=self.data
+            )
+            assert response.status_code == 200, response.content
+
+    def test_owner_update_allowed_with_correct_token_scope(self):
         self.create_member(
             user=self.user,
             organization=self.project.organization,
-            teams=[self.project.teams.first()],
-            role="admin",
+            teams=[self.team],
+            role="owner",
         )
 
-        token = ApiToken.objects.create(user=self.user, scope_list=["project:write"])
-        authorization = f"Bearer {token.token}"
+        for scope in ["project:write", "project:admin"]:
+            authorization = self.create_bearer_token(self.user, [scope])
 
-        data = {"platform": "rust"}
+            response = self.client.put(
+                self.url, format="json", HTTP_AUTHORIZATION=authorization, data=self.data
+            )
+            assert response.status_code == 200, response.content
 
-        url = reverse(
-            "sentry-api-0-project-details",
-            kwargs={
-                "organization_slug": self.project.organization.slug,
-                "project_slug": self.project.slug,
-            },
+    def test_member_update_denied_with_token(self):
+        self.create_member(
+            user=self.user,
+            organization=self.project.organization,
+            teams=[self.team],
+            role="member",
         )
-        response = self.client.put(url, format="json", HTTP_AUTHORIZATION=authorization, data=data)
-        assert response.status_code == 200, response.content
 
-    def test_admin_updates_denied_with_token(self):
+        authorization = self.create_bearer_token(self.user, ["project:write"])
```

Comment:
is it even possible for a member to have `project:write` on a token? i thought users couldn't have more scopes than their lowest org role across all organizations or something

Reply:
It's possible for a user to create a user token with any possible scopes because they could have different org roles in their orgs with varying levels of permissions.